crates/solidity/testing/solc/src/commands/dissect.rs

```diff
@@ -0,0 +1,172 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use clap::Parser;
+use console::{style, Color};
+use infra_utils::paths::PathExtensions;
+use infra_utils::terminal::Terminal;
+use itertools::Itertools;
+use semver::Version;
+use solidity_language::SolidityDefinition;
+
+use crate::utils::{
+    ApiInput, Binary, Error, InputSource, LanguageSelector, Severity, SourceLocation,
+};
+
+#[derive(Debug, Parser)]
+pub struct DissectCommand {
+    /// File path to the Solidity (*.sol) source file to parse
+    file: PathBuf,
+}
+
+impl DissectCommand {
+    pub fn execute(self) -> Result<()> {
+        let language = SolidityDefinition::create();
+        let binaries = Binary::fetch_all(&language)?;
+
+        let mut state = State::new(self.file.to_owned())?;
+
+        for binary in binaries {
+            let input = InputSource {
+                content: state.source.to_owned(),
+            };
+
+            let output = binary.run(&ApiInput {
+                language: LanguageSelector::Solidity,
+                sources: [(self.file.unwrap_string(), input)].into(),
+            });
+
+            let errors = match output {
+                Ok(output) => output.errors.unwrap_or_default(),
+                Err(error) => vec![Error {
+                    message: format!("{error}"),
+                    severity: Severity::Error,
+                    location: None,
+                }],
+            };
+
+            state.apply(binary.version.to_owned(), errors)?;
+        }
+
+        state.print_all()?;
+
+        Ok(())
+    }
+}
+
+struct State {
+    file: PathBuf,
+    source: String,
+
+    errors: Vec<Error>,
+    versions: Vec<Version>,
```

Comment:
Similarly, just a note or a rename would help understand the code better (e.g. are those already tested versions? are these only versions that errored out? are these versions yet to test?)

Reply:
Thanks. I refactored the module a little bit to make it clearer, and added some comments. Please let me know what you think.